File: Solutions/_3.cs
public class Solution
{
    public int LengthOfLongestSubstring(string s)
    {
        if (s.Length == 1)
        {
            return 1;
        }
        Hashtable hashtable = new Hashtable();
        int max = 0;
        int start = 0;
        for (int i = 0; i < s.Length; i++)
        {
            if (hashtable.ContainsKey(s[i]))
            {
                start = Math.Max(start, (int)hashtable[s[i]] + 1);
            }
            max = Math.Max(max, i - start + 1);
            hashtable[s[i]] = i;
        }
        return max;
    }
}

File: Solutions/_3.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    const map = new Map();
    let max = 0;
    let start = 0;
    for (let i = 0; i < s.length; i++) {
        const char = s[i];
        if (map.has(char)) {
            start = Math.max(start, map.get(char) + 1);
        }
        max = Math.max(max, i - start + 1);
        map.set(char, i);
    }
    return max;
};